### PR TITLE
Daily supporter churn email

### PR DESCRIPTION
**Pull request by johnduffell**
So at present we have a KR stating to "_Have renewal, payment failure and supporter satisfaction metrics which we (the Guardian) can measure and agree on_"

Apparently this is doing well, but to gain trust that we are getting value from this, we need visibility amongst the KR team (and onwards to those who will use it).

As such I have implemented a minimum viable product to demonstrate this, which should produce a headline figure every day for supporter churn.  If that is happening and everyone is agreeing (and ideally wanting improvements/adding new people) then I would have good confidence that the metric is useful.  In that case we should be giving a high confidence of meeting that part of the OKR.

Initially this email should go out early in the morning, but based on feedback this will probably change.

Example email:

image

@pvighi @paulbrown1982 @lmath @AWare @jacobwinch

### CODE REVIEW

File: cfn.yaml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    Type: String
+    Default: supporter-metrics-job-dist
+  EmailFrom:
+    Description: from address for emails
+    Type: String
+    Default: john.duffell@guardian.co.uk
+  EmailTo:
+    Description: email address to send emails to
+    Type: String
+    Default: john.duffell@guardian.co.uk
```
Comment (johnduffell) [inline cfn.yaml:30]: in CODE I will send to me, but in prod I will send to the KR team.

File: cfn.yaml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    Description: Stage name
+    Type: String
+    AllowedValues:
+      - CODE
+      - PROD
+    Default: CODE
+  DeployBucket:
+    Description: Bucket where RiffRaff uploads artifacts on deploy
+    Type: String
+    Default: supporter-metrics-job-dist
```
Comment (johnduffell) [inline cfn.yaml:22]: I wasn't sure if I should use the membership-dist bucket or a new one?

File: project/plugins.sbt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.4")
+
+addSbtPlugin("com.gu" % "sbt-riffraff-artifact" % "0.9.8")
+
+resolvers += Resolver.url(
+  "idio",
+  url("http://dl.bintray.com/idio/sbt-plugins")
+)(Resolver.ivyStylePatterns)
+
+addSbtPlugin("org.idio" % "sbt-assembly-log4j2" % "0.1.0")
```
Comment (johnduffell) [inline project/plugins.sbt:10]: this is needed as it seems presto and lambda-log4j both pull in a dat file which needs a special merge.

File: cfn.yaml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        S3Key: !Sub ${Stack}/${Stage}/${App}/${App}.jar
+      Environment:
+        Variables:
+          Stage: !Ref Stage
+          Stack: !Ref Stack
+          App: !Ref App
+          EmailFrom: !Ref EmailFrom
+          EmailTo: !Ref EmailTo
+          PrestoUrl: !Ref PrestoUrl
+          EmailLine: !Ref EmailLine
```
Comment (johnduffell) [inline cfn.yaml:91]: this way of passing private/config to the lambda is only sussatinable for small numbers of params.  We might need an S3 file if/when it gets too much

**Comment by paulbrown1982** [comment]
👍 awesome MVP!

**Comment by jacobwinch** [comment]
This is ace :1st_place_medal:

**Pull request merged**